packages/ssr-web/app/routes/index.ts

```diff
@@ -0,0 +1,82 @@
+import Route from '@ember/routing/route';
+import '../css/pay.css';
+import { inject as service } from '@ember/service';
+import * as Sentry from '@sentry/browser';
+import AppContextService from '@cardstack/ssr-web/services/app-context';
+import config from '../config/environment';
+import { generateMerchantPaymentUrl, gqlQuery } from '@cardstack/cardpay-sdk';
+
+interface UserSpaceRouteModel {
+  id: string;
+  name: string;
+  backgroundColor: string;
+  textColor: string;
+  paymentURL: string;
+}
+
+export default class UserSpaceRoute extends Route {
+  @service('app-context') declare appContext: AppContextService;
+
+  async model(): Promise<UserSpaceRouteModel> {
+    if (this.appContext.currentApp === 'card-space') {
+      try {
+        const cardSpaceResult: {
+          included: any[];
+        } = await (
+          await fetch(
+            `${config.hubURL ?? 'http://localhost:3000'}/api/card-spaces/${
+              this.appContext.cardSpaceId
+            }`,
+            {
+              method: 'GET',
+              headers: {
+                Accept: 'application/vnd.api+json',
+                'Content-Type': 'application/vnd.api+json',
+              },
+            }
+          )
+        ).json();
+        let merchant = cardSpaceResult.included.find(
+          (v) => v.type === 'merchant-infos'
+        );
+        let queryResult = await gqlQuery(
+          config.chains.layer2,
+          `query($did: String!) {
+          merchantSafes(where: { infoDid: $did }) {
+            id
+          }
+        }`,
+          {
+            did: merchant.attributes.did,
+          }
+        );
+        let address = queryResult.data.merchantSafes[0].id;
+
+        if (!merchant || !address) {
+          // TODO: replace with proper 404 somehow
+          // this 404 is for card space
```

Comment:
The app context service could be used to inform the 404 about what to show on the 404 page.